### DISCUSSION TITLE
Coder expertise need to fix python error.

### DISCUSSION THREAD

**Discussion by DrSupermanShadow** [Bforartists/Bforartists#2909] [General]
def invoke(self, context, event):
        dpi_value = Common.get_user_preferences().system.dpi
        return context.window_manager.invoke_props_dialog(self, width=dpi_value * 3.25)
        
        What is the new correct python 310 script code to correct this line error?
        keep it simple. this python correction notice is way too complex for sum 1 with autism.
        Since pro-coders refuse to correct the plug-in in a timely manner i am going to fix the script codes myself.
        after all its an open-source plugin
        
        
        Info: Saved text "C:\Program Files\SS ART UI\3.2\scripts\addons\cats-blender-plugin-master\ui\armature.py"

Python: Traceback (most recent call last):
  File "C:\Program Files\SS ART UI\3.2\scripts\addons\cats-blender-plugin-master\ui\armature.py", line 205, in invoke
    return context.window_manager.invoke_props_dialog(self, width=dpi_value * 3.25)
TypeError: WindowManager.invoke_props_dialog(): error with keyword argument "width" -  Function.width expected an int type, not float

location: <unknown location>:-1
Error: Python: Traceback (most recent call last):
  File "C:\Program Files\SS ART UI\3.2\scripts\addons\cats-blender-plugin-master\ui\armature.py", line 205, in invoke
    return context.window_manager.invoke_props_dialog(self, width=dpi_value * 3.25)
TypeError: WindowManager.invoke_props_dialog(): error with keyword argument "width" -  Function.width expected an int type, not float

**Comment by ghost** [comment]
You just need to convert `dpi_value * 3.25` to integer, like this

`width=int(dpi_value*3.25)`

Also note this isn't a Bforartists issue so it doesn't belong here😄

There's also this server for coding help unrelated to Bforartists

https://discord.gg/CvFYAuQcDT

**Reply by DrSupermanShadow** [reply]
Discord doesn't allow people without contracted cellphone services to use their service.
 So they basically are taking part in classism, and disability discrimination. Since there are a lot of disabled & nonpeople who have to use prepaid, google voice or VoIP-based services for phone service... sum people can only afford land lines.
 If your account get flagged for phone verification.
They refuse to override it.
For example, if you have a unique name that doesn't sound real..
 like Thor Asgardian( ys there is people in the world  with the first name Thor and weird last name that don't sound like real last names)>>
 
 any whoos thanks forhow to read that code line but the issue isn't resolved, why can't python say that code error fix, 
 I actually understood that **b for artist python version code error change.**..
 
 ... and I am gonna have to disagree with you.
 since the script for the  Cats plugin 
 
 **is USING B for artists Python edition.**
 &  **B for artist** won't run without that **python edition.**
 & you have to fix the script for Cats **within B For artists Python Edition**
 & ADDon only work with what is in the B for artists python version of  script coding
 That's a whole lot of your program being used even if B for artists didnt design the  Cats Plugin.
 OR PYTHON PROGRAM
 
 And script errors count as bugs. especially if the fix listing is not readable...
 
In the autism world if a program for example like B for artists Python programs 
lists an error code and its fix as something that is unintelligible and doesn't provide a clear example like you just provided even after watching all the tutorials they count that as a bug.
 
 So you may want to fix B for the artists' PYTHON Version so the error codes and fix are actually readable...
 and from what I've been able to gather from experimentation. 
 the python coding and UI visuals can only be coded to match the program like b for artists. so i have to use the specific UI version that came with your program...
 
 But thanks for trying to help....

**Reply by ReinerBforartists** [reply]
I know it is hard for you to understand, but you must understand that Bforartists is just the base software here. Addons resides on top of it. Addons comes usually from third party members. Not from us. Fixing bugs in these addons is not our job. We did not create these addons. And we are not responsible for them.

When an Addon throws an error, then the Addon creator did something wrong. Not we. And we are also not responsible to fix this error then.

It is a different story is when we talk about the addons that ships with Blender / Bforartists. But the addons that you try to fix here are not included in Bforartists. It is an external extra download. Please contact the addon author for bug fixes. We simply cannot help here.

**Comment by ReinerBforartists** [comment]
Yes. Such issues does not belong into the Bforartists tracker. As much as i would love to, we simply miss the man power to fix addons too.

This is best asked at the forums. Or as mentioned by Iyad Ahmed in the Discord server. I will move it to the discussions board therefore.

The very best place to ask for help is the maker of this addon though. It's his responsibility. And he should be interested in fixing bugs too. Of course just in case he still develops at it.

Kind regards
Reiner
